pkg/stream/edgedlogconnection.go

```diff
@@ -18,128 +18,77 @@ package stream
 
 import (
 	"bufio"
-	"encoding/json"
-	"errors"
 	"fmt"
 	"io"
 	"net/http"
-	"net/url"
 
 	"k8s.io/klog/v2"
 )
 
 type EdgedLogsConnection struct {
-	ReadChan chan *Message `json:"-"`
-	Stop     chan struct{} `json:"-"`
-	MessID   uint64        // message id
-	URL      url.URL       `json:"url"`
-	Header   http.Header   `json:"header"`
-}
-
-func (l *EdgedLogsConnection) GetMessageID() uint64 {
-	return l.MessID
-}
-
-func (l *EdgedLogsConnection) CacheTunnelMessage(msg *Message) {
-	l.ReadChan <- msg
-}
-
-func (l *EdgedLogsConnection) CloseReadChannel() {
-	close(l.ReadChan)
-}
-
-func (l *EdgedLogsConnection) CleanChannel() {
-	for {
-		select {
-		case <-l.Stop:
-		default:
-			return
-		}
-	}
+	BaseEdgedConnection `json:",inline"`
 }
 
 func (l *EdgedLogsConnection) CreateConnectMessage() (*Message, error) {
-	data, err := json.Marshal(l)
-	if err != nil {
-		return nil, err
-	}
-	return NewMessage(l.MessID, MessageTypeLogsConnect, data), nil
+	return l.createConnectMessage(MessageTypeLogsConnect, l)
 }
 
 func (l *EdgedLogsConnection) String() string {
 	return fmt.Sprintf("EDGE_LOGS_CONNECTOR Message MessageID %v", l.MessID)
 }
 
-func (l *EdgedLogsConnection) receiveFromCloudStream(stop chan struct{}) {
+func (l *EdgedLogsConnection) receiveFromCloudStream() {
```

Comment:
Can we use other name? Such as `edgedlog` instead of `l`? Single-letter variables are somewhat irregular

Reply:
Done.

Reply:
Golang prefers concise variable name definitions, so I changed it to `el`. 😊